### ISSUE TITLE
Link label with ^ not recognized

### ISSUE THREAD

**Issue by villeodell** [DavidAnson/markdownlint#792]
From <https://spec.commonmark.org/0.30/#link-label>:

> A [link label](https://spec.commonmark.org/0.30/#link-label) begins with a left bracket ([) and ends with the first right bracket (]) that is not backslash-escaped. Between these brackets there must be at least one character that is not a space, tab, or line ending. Unescaped square bracket characters are not allowed inside the opening and closing square brackets of [link labels](https://spec.commonmark.org/0.30/#link-label). A link label can have at most 999 characters inside the square brackets.

> One label [matches](https://spec.commonmark.org/0.30/#matches) another just in case their normalized forms are equal. To normalize a label, strip off the opening and closing brackets, perform the Unicode case fold, strip leading and trailing spaces, tabs, and line endings, and collapse consecutive internal spaces, tabs, and line endings to a single space. If there are multiple matching reference link definitions, the one that comes first in the document is used. (It is desirable in such cases to emit a warning.)

The following example is parsed by commonmark.js ([see here](https://spec.commonmark.org/dingus/?text=%5B%5E1%5D%0A%0A%5B%5E1%5D%3A%20%2Furl))

```markdown
[^1]

[^1]: /url
```

However, markdownlint complains with 

`No link definition found: '^1'(link.no-such-reference)`

In addition to the above reason for supporting, support for these types of link labels would also be a particular help for users of [footnotes](https://github.blog/changelog/2021-09-30-footnotes-now-supported-in-markdown-fields/).

**Comment by villeodell** [comment]
My apologies! I thought this was markdownlint complaining, when in fact it's actually vscode 🙄 .

https://github.com/DavidAnson/vscode-markdownlint/issues/262

https://github.com/microsoft/vscode/issues/162691

**Comment by DavidAnson** [comment]
No worries! I hope they fix this soon. :)
